README.md

```diff
@@ -80,7 +78,9 @@ Rough outline of a possible resolution algorithm:
 
 In the future, the algorithm might be adjusted to align with work done in the [import maps proposal](https://github.com/domenic/import-maps).
 
-### Usage
+For packages that only have a main and no exports, `"exports": false` can be used as a shorthand for `"exports": {}` providing an encapsulated package.
```

Comment:
cc @GeoffreyBooth i think this exact use case is a reason it’s useful, in that other thread, to have exports override whatever main points to, instead of providing main itself.

Reply:
Can you elaborate? This suggests having: `{"main": "./foo.mjs", "exports": false}` which would break if `exports` would completely overwrite `main`..? It does work if it desugars to merge semantics (`Object.assign({ ".": mainValue || [] }, exports)`).

Reply:
that’s a good point, but I’m more thinking conceptually.

Reply:
> i think this exact use case is a reason it’s useful, in that other thread, to have exports override whatever main points to, instead of providing main itself.

I’m not sure I follow; you mean the reason that `exports: false` is useful?

A package needs to export _something,_ right? `exports: false` and no `main` would be the package equivalent of dead code?

I think I agree with @jkrems that merge semantics make the most sense for this, since overriding `main` with `exports: false` would have the effect of creating an unusable package in all versions of Node that support `exports`. We’re obviously not dropping support for `main`, so if `exports` allows for setting the main as well it would just take precedence.

Reply:
i don’t mean that. I mean that exports has the ability to override whatever main points to. exports false wouldn’t be using that ability.

Reply:
It’s a question of semantics. The docs could simply define `exports: false` as “no exports other than whatever is defined in `"main"`.” or something like that. That’s probably what they _should_ say.

Because if you think about it, the _lack_ of `"."` in `exports` wouldn’t mean that there’s no main export defined for the package. I.e. we’d want to support:

```js
"main": "./main.js",
"exports": { "./foo": "./foo/index.js" }
```

Where `require('pkg')` and `require('pkg/foo')` both work. This would be an example of the merging/`Object.assign` approach, rather than `exports` overriding `main` in the sense that this package would not have any main export defined because there’s no `"."` in `exports`.

Reply:
It sounds like you both support the same semantics and the only disagreement is how it should be worded? My vote is for calling it "merge" because as @GeoffreyBooth points out, it's only the `"."` key that actually overrides anything. But I think everyone agrees that we should bring back dot-main in some form (the ability to specify `main` in `exports` in a way to wins over the top-level `main` field).

Reply:
Perhaps `"."` should have its own small section in the docs. Something like:

> `"."` can specify the root entry point for a package, i.e. the file loaded for `require('pkg')` or `import 'pkg'`. Historically this has been defined by the `package.json` `"main"` field, and `"main"` will continue to be supported. If `"main"` is defined and `"."` is not, the value in `"main"` will determine the package root entry point; if both `"main"` and `"."` are defined, the latter takes precedence. If `"main"` is defined and `exports` is set to `false`, the value in `"main"` is the only export available for the package.

Since `"main"` doesn’t define an object it’s a little weird to discuss merging at all. We’re really only talking about the precedence of `"main"` versus `"."`.

Reply:
Hopefully we can discuss this main problem at the coming meeting.

I would ideally like to get the same spec here merged so that we can also start to discuss it further though, and wouldn't want these main considerations to block that.

Is there something we can do here to resolve this further? Any note or clarifications?

Reply:
I don't think there's any disagreement? It seems like we're just discussing the best way to explain the behavior to users, which is something we can worry about when we get to writing the docs.